system/qcomgpsd/structs.py

```diff
@@ -290,32 +291,42 @@ def name_to_camelcase(nam):
       i += 1
   return ''.join(ret)
 
+# MappingProxyType is used to make this dictionary read-only.
+TYPES = MappingProxyType({
+    "double": "d",
+    "float": "f",
+    "int8": "b",
+    "int8_t": "b",
+    "uint8": "B",
+    "uint8_t": "B",
+    "int16": "h",
+    "int16_t": "h",
+    "uint16": "H",
+    "uint16_t": "H",
+    "int32": "i",
+    "int32_t": "i",
+    "uint32": "I",
+    "uint32_t": "I",
+    "int64": "q",
+    "int64_t": "q",
+    "uint64": "Q",
+    "uint64_t": "Q",
+  })
+
 def parse_struct(ss):
   st = "<"
   nams = []
-  for l in ss.strip().split("\n"):
-    if len(l.strip()) == 0:
```

Comment:
For speed up PRs, please include a benchmark of the before and after performance